lib/mongoose_push/api/v1/response_encoder.ex

```diff
@@ -0,0 +1,14 @@
+defmodule MongoosePush.API.V1.ResponseEncoder do
+  @moduledoc """
+  Module for handling internal responses to V1 HTTP2 codes
+  """
+  @behaviour MongoosePush.API
+  alias MongoosePush.Service
+
+  @spec to_status(:ok | {:error, Service.error()} | {:error, MongoosePush.error()}) ::
+          {non_neg_integer, %{details: atom | String.t()} | nil}
```

Comment:
This spec is invalid - depending on API version, the key may be different. We should probably relax that to just a map.